File: src/main/java/shape/Line.java
package shape;

public class Line {
    private Point pointOne;
    private Point pointTwo;

    public Line(Point pointOne, Point pointTwo) {
        this.pointOne = pointOne;
        this.pointTwo = pointTwo;


        // Check if the lenght is 0, if so throw an exception
        if (pointOne.GetX() == pointTwo.GetX() && pointOne.GetY() == pointTwo.GetY()) {
            throw new IllegalArgumentException("The two points cannot be the same.");
        }
    }

    public Point GetPointOne() {
        return pointOne;
    }

    public Point GetPointTwo() {
        return pointTwo;
    }

    public double GetSlope() {
        double slope = (pointTwo.GetY() - pointOne.GetY()) / (pointTwo.GetX() - pointOne.GetX());
        return slope;
    }

    public double GetLength() {
        double length = Math.sqrt(Math.pow(pointTwo.GetX() - pointOne.GetX(), 2) + Math.pow(pointTwo.GetY() - pointOne.GetY(), 2));
        return length;
    }

    public void MoveLine(double addXvalue, double addYvalue) {
        pointOne.MovePoint(addXvalue, addYvalue);
        pointTwo.MovePoint(addXvalue, addYvalue);
    }
}

File: src/main/kotlin/shape/Line.kt
package shape

import kotlin.math.hypot

class Line(private val pointOne: Point, private val pointTwo: Point) {

    init {
        if (pointOne.GetX() == pointTwo.GetX() &&
            pointOne.GetY() == pointTwo.GetY()
        ) {
            throw IllegalArgumentException("The two points cannot be the same.")
        }
    }

    fun GetPointOne(): Point {
        return pointOne
    }

    fun GetPointTwo(): Point {
        return pointTwo
    }

    fun GetSlope(): Double {
        return (pointTwo.GetY() - pointOne.GetY()) /
                (pointTwo.GetX() - pointOne.GetX())
    }

    fun GetLength(): Double {
        return hypot(
            pointTwo.GetX() - pointOne.GetX(),
            pointTwo.GetY() - pointOne.GetY()
        )
    }

    fun MoveLine(addXvalue: Double, addYvalue: Double) {
        pointOne.MovePoint(addXvalue, addYvalue)
        pointTwo.MovePoint(addXvalue, addYvalue)
    }
}